### ISSUE TITLE
OnSnapshot(useFirestoreDoc) is returning result twice. 1. fromCache: false 2. fromCache: true

### ISSUE THREAD

**Issue by catrawalkar** [FirebaseExtended/reactfire#188]
### Version info

**React:** 16.9.0

**Firebase:** 7.2.0

**ReactFire:** 2.0.0-canary.1fce6b9

**Other (e.g. Node, browser, operating system) (if applicable):**

### Issue

When I use **useFirestoreDoc**, it has useState hook somewhere inside (I am guessing somewhere inside **useObservable**) and **useFirestoreDoc** is using onSnapshot method of firebase and returns **snapshot** result of queried document.

When I tried to log this snapshot, I am getting the result twice: 
`1. _fromCache: false`
`2. _fromCache: true`

This renders the component 2 times, because useState hook is used inside useObservable.

### My Questions

- Why it is sending 2 results??

- How can I rectify this 2 renders issue?

- `_fromCache: true` is only expected when user is offline, why it is fetching from cache even when user is online??

**Comment by catrawalkar** [comment]
![image](https://user-images.githubusercontent.com/15074162/69333502-bcbae900-0c7e-11ea-806e-16b4bb58e0d4.png)

for eg, in here the console.log shows data was returned 4 times. 2 times for `fromCache: false` then 2 times for `fromCache: true`.

**Comment by catrawalkar** [comment]
Sorry, If I was not clear..

Tldr:

OnSnapshot gets called twice..! So returns 2 values..and thus component renders twice

**Comment by jhuleatt** [comment]
Thanks for the issue @catrawalkar. I'll take a look!

**Comment by jhuleatt** [comment]
I haven't had much time to look at this. Hopefully will have more this week. My assumption that this is caused by `useObservable` piping an initial value into the Observable: https://github.com/FirebaseExtended/reactfire/blob/e1daab435f10c629267ff8c2966da878327c941d/reactfire/useObservable/index.ts#L68

**Comment by catrawalkar** [comment]
I am unable to understand what piping means or what .pipe function is doing.. can you please refer me to some link where i can read more about this and may be send a pull request.

**Comment by jhuleatt** [comment]
ReactFire uses [RxFire](https://firebase.googleblog.com/2018/09/introducing-rxfire-easy-async-firebase.html) under the hood, which is based on [RxJS](https://rxjs.dev/) Observables. Piping lets you apply operators to observables: https://rxjs-dev.firebaseapp.com/guide/operators#piping

**Comment by catrawalkar** [comment]
@jhuleatt

**Comment by catrawalkar** [comment]
Am unable to understand this :( sorry @jhuleatt

**Comment by jamesdaniels** [comment]
This is the expected behavior of the Firebase JS SDK, which we use. Feature request is #195
